src/app/bulk-edit/bulk-edit-select/bulk-edit-select.component.ts

```diff
@@ -131,18 +188,111 @@ export class BulkEditSelectComponent implements OnInit, OnDestroy {
-        catchError((error) => {
-          this.messageHandler.showError(
-            'There was a problem finding the Data Elements.',
-            error
-          );
-          return EMPTY;
-        })
+  private resetContext() {
+    this.context.childDomainType = null;
+    this.context.childItems = [];
+    this.context.profiles = [];
+    this.contextChanged.emit(this.context);
+  }
+
+  private getSuitableChildDomainTypes(): CatalogueItemDomainTypeOption[] {
+    if (
+      this.context.rootItem.domainType === CatalogueItemDomainType.DataModel
+    ) {
+      return [
+        {
+          domainType: CatalogueItemDomainType.ModelDataType,
+          displayName: 'Data types'
+        },
+        {
+          domainType: CatalogueItemDomainType.DataClass,
+          displayName: 'Data classes'
+        },
+        {
+          domainType: CatalogueItemDomainType.DataElement,
+          displayName: 'Data elements'
+        }
+      ];
+    }
+
+    if (
+      this.context.rootItem.domainType === CatalogueItemDomainType.DataClass
+    ) {
+      return [
+        {
+          domainType: CatalogueItemDomainType.DataElement,
+          displayName: 'Data elements'
+        }
+      ];
+    }
+
+    if (
+      this.context.rootItem.domainType ===
+        CatalogueItemDomainType.Terminology ||
+      this.context.rootItem.domainType === CatalogueItemDomainType.CodeSet
+    ) {
+      return [
+        {
+          domainType: CatalogueItemDomainType.Term,
+          displayName: 'Terms'
+        }
+      ];
+    }
+
+    return [];
+  }
+
+  private loadChildItems(): Observable<MauroItem[]> {
+    let request$: Observable<MdmIndexResponse<MauroItem>>;
+    const filters: FilterQueryParameters = { all: true, sort: 'label' };
+
+    if (this.context.childDomainType === CatalogueItemDomainType.DataElement) {
+      request$ =
+        this.context.rootItem.domainType === CatalogueItemDomainType.DataClass
+          ? this.resources.dataElement.list(
+              this.context.rootItem.model,
+              this.context.rootItem.id,
+              filters
+            )
+          : this.resources.dataModel.dataElements(
+              this.context.rootItem.id,
+              filters
+            );
+    } else if (
+      this.context.childDomainType === CatalogueItemDomainType.DataClass
+    ) {
+      request$ = this.resources.dataClass.all(
+        this.context.rootItem.id,
+        filters
+      );
+    } else if (
+      this.context.childDomainType === CatalogueItemDomainType.ModelDataType
+    ) {
+      request$ = this.resources.dataType.list(
+        this.context.rootItem.id,
+        filters
       );
+    } else if (this.context.childDomainType === CatalogueItemDomainType.Term) {
+      request$ =
+        this.context.rootItem.domainType === CatalogueItemDomainType.CodeSet
+          ? this.resources.codeSet.terms(this.context.rootItem.id, filters)
+          : this.resources.term.list(this.context.rootItem.id, filters);
```

Comment:
This might need paginating

Reply:
Need to look into this one, did not know that the `all` param doesn't work for term lists.

Reply:
I've temporarily worked around this by doing an initial list request to get the total count, then doing a second list request with `max: count`. It could be improved at a later date.